### PR TITLE
Fix types in documentation

### PR DESCRIPTION
**Pull request by ivanpopelyshev**
Its #5317 without auto-types

### CODE REVIEW

File: bundles/pixi.js/src/index.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ loaders.Loader.registerPlugin(textBitmap.BitmapFontLoader);
 loaders.Loader.registerPlugin(spritesheet.SpritesheetLoader);
 
 app.Application.registerPlugin(ticker.TickerPlugin);
-app.Application.registerPlugin(loaders.LoaderPlugin);
+app.Application.registerPlugin(loaders.AppLoaderPlugin);
```
Comment (englercj) [inline bundles/pixi.js/src/index.js]: This looks like a functional change, is this intended?

File: packages/canvas/canvas-sprite/src/CanvasTinter.js [hunk 2]
```diff
@@ -274,21 +274,14 @@ const CanvasTinter = {
...
      *
      * @memberof PIXI.CanvasTinter
-     * @type {tintMethodFunctionType}
+     * @type {Function}
      */
-    tintMethod: 0,
+    tintMethod: () =>
+    { // jslint-disable no-empty-function
+
+    },
```
Comment (englercj) [inline packages/canvas/canvas-sprite/src/CanvasTinter.js]: Why did this change from using the callback typedef?

File: packages/core/src/AbstractRenderer.js [hunk 3]
```diff
@@ -118,13 +118,6 @@ export default class AbstractRenderer extends EventEmitter
         this.autoDensity = options.autoDensity || options.autoResize || false;
         // autoResize is deprecated, provides fallback support
 
-        /**
-         * Tracks the blend modes useful for this renderer.
-         *
-         * @member {object<string, mixed>}
-         */
-        this.blendModes = null;
```
Comment (englercj) [inline packages/core/src/AbstractRenderer.js]: Is this removal intentional?

File: packages/core/src/filters/FilterSystem.js [hunk 4]
```diff
@@ -330,8 +331,8 @@ export default class FilterSystem extends System
      * Draws a filter.
      *
      * @param {PIXI.Filter} filter - The filter to draw.
-     * @param {PIXI.RenderTarget} input - The input render target.
-     * @param {PIXI.RenderTarget} output - The target to output to.
+     * @param {PIXI.RenderTexture} input - The input render target.
+     * @param {PIXI.RenderTexture} output - The target to output to.
```
Comment (englercj) [inline packages/core/src/filters/FilterSystem.js]: Are these RenderTarget -> RenderTexture changes correct? In the past these have been different things.

File: packages/core/src/filters/filterTransforms.js [hunk 5]
```diff
@@ -4,10 +4,9 @@ import { Matrix } from '@pixi/math';
  * Calculates the mapped matrix
  * @param filterArea {Rectangle} The filter area
  * @param sprite {Sprite} the target sprite
- * @param outputMatrix {Matrix} @alvin
+ * @param outputMatrix {Matrix} matrix that will normalize map filter cords in the filter to screen space
```
Comment (englercj) [inline packages/core/src/filters/filterTransforms.js]: These should be reformatted to `@param {type} name`

File: packages/core/src/framebuffer/Framebuffer.js [hunk 6]
```diff
@@ -6,7 +6,7 @@ import { FORMATS, TYPES } from '@pixi/constants';
  * @class
  * @memberof PIXI
  */
-export default class FrameBuffer
+export default class Framebuffer
```
Comment (englercj) [inline packages/core/src/framebuffer/Framebuffer.js]: Unrelated to this change, I thought we weren't using `export default` anymore?

File: packages/core/src/geometry/Attribute.js [hunk 7]
```diff
@@ -8,7 +8,7 @@
  * @class
  * @memberof PIXI
  */
-class Attribute
+export default class Attribute
```
Comment (englercj) [inline packages/core/src/geometry/Attribute.js]: I think these exports should be named exports, IIRC we decided to not do export default whenever possible.

File: packages/core/src/index.js [hunk 8]
```diff
@@ -28,6 +29,7 @@ export { default as QuadUv } from './utils/QuadUv';
 export { default as checkMaxIfStatementsInShader } from './shader/utils/checkMaxIfStatementsInShader';
 export { default as Shader } from './shader/Shader';
 export { default as Program } from './shader/Program';
+export { default as GLProgram } from './shader/GLProgram';
 export { default as UniformGroup } from './shader/UniformGroup';
 export { default as SpriteMaskFilter } from './filters/spriteMask/SpriteMaskFilter';
 export { default as Filter } from './filters/Filter';
```
Comment (englercj) [inline packages/core/src/index.js]: Wow, I must really be off-base about default exports or something changed recently?

File: packages/core/src/textures/resources/BufferResource.js [hunk 9]
```diff
@@ -1,5 +1,9 @@
 import Resource from './Resource';
 
+/**
+ * @interface SharedArrayBuffer
+ */
```
Comment (englercj) [inline packages/core/src/textures/resources/BufferResource.js]: Is this needed?

File: packages/graphics/src/utils/Star.js [hunk 10]
```diff
@@ -5,6 +5,7 @@ import { Polygon, PI_2 } from '@pixi/math';
  *
  * @class
  * @extends PIXI.Polygon
+ * @memberOf PIXI
```
Comment (englercj) [inline packages/graphics/src/utils/Star.js]: `@memberof` is the correct tag.

File: packages/core/src/textures/GLTexture.js [hunk 11]
```diff
@@ -1,10 +1,14 @@
+/**
+ * Internal texture for WebGL context
+ * @class
+ * @memberOf PIXI
```
Comment (englercj) [inline packages/core/src/textures/GLTexture.js]: `@memberof` is the correct tag.

File: packages/math/src/ObservablePoint.js [hunk 12]
```diff
@@ -141,3 +141,9 @@ export default class ObservablePoint
         }
     }
 }
+
+/**
+ * A number, or a string containing a number.
+ * @memberOf PIXI
```
Comment (englercj) [inline packages/math/src/ObservablePoint.js]: `@memberof` is the correct tag.

File: packages/particles/src/ParticleShader.js [hunk 13]
```diff
@@ -1,21 +0,0 @@
...
-export default class ParticleShader extends GLShader
-{
-    /**
-     * @param {PIXI.Shader} gl - The webgl shader manager this shader works for.
-     */
-    constructor(gl)
-    {
-        super(gl, vertex, fragment, PRECISION.DEFAULT);
-    }
-}
```
Comment (englercj) [inline packages/particles/src/ParticleShader.js]: Was this deletion intentional?

File: packages/core/src/AbstractRenderer.js (hunk 3 above)
Comment (ivanpopelyshev) [reply packages/core/src/AbstractRenderer.js]: its not used in webgl anymore

File: packages/core/src/filters/FilterSystem.js (hunk 4 above)
Comment (ivanpopelyshev) [reply packages/core/src/filters/FilterSystem.js]: RenderTexture/RenderTarget functionality was moved to RenderTexture/Framebuffer

File: packages/core/src/textures/resources/BufferResource.js (hunk 9 above)
Comment (ivanpopelyshev) [reply packages/core/src/textures/resources/BufferResource.js]: Yes, unfortunately we cannot ask users to add es2017 lib to their tsconfig yet.

File: packages/particles/src/ParticleShader.js (hunk 13 above)
Comment (ivanpopelyshev) [reply packages/particles/src/ParticleShader.js]: Unused.

File: packages/canvas/canvas-sprite/src/CanvasTinter.js (hunk 2 above)
Comment (ivanpopelyshev) [reply packages/canvas/canvas-sprite/src/CanvasTinter.js]: There was a callback? I just removed `tintMethodFunctionType`

File: bundles/pixi.js/src/index.js (hunk 1 above)
Comment (ivanpopelyshev) [reply bundles/pixi.js/src/index.js]: We have to separate ILoaderPlugin and AppLoaderPlugin. We have `LoaderPlugin` that is actually application plugin.

**Comment by ivanpopelyshev** [comment]
We want to show @protected in types but not in docs, however we have a problem with it: `opts { access: 'public' }` removes everything.

File: packages/canvas/canvas-sprite/src/CanvasTinter.js (hunk 2 above)
Comment (englercj) [reply packages/canvas/canvas-sprite/src/CanvasTinter.js]: Yeah the tag is `@callback` which defines a function typedef. I'm asking why you removed that and replaced it with a `Function` type.

File: bundles/pixi.js/src/index.js (hunk 1 above)
Comment (bigtimebuddy) [reply bundles/pixi.js/src/index.js]: @ivanpopelyshev and I discussed this. I'm find with that change.
